popmon/visualization/utils.py

```diff
@@ -349,20 +351,21 @@ def plot_overlay_1d_histogram_b64(
                 width = np.diff(bin_edges)
 
             # plot histogram
-            ax.bar(
-                bin_edges[:-1],
-                bin_values,
-                width=width,
-                alpha=alpha,
-                label=hist_names[i],
+            fig.add_trace(
+                go.Bar(
+                    x=bin_edges[1:],
+                    y=bin_values,
+                    showlegend=True,
+                    opacity=alpha,
+                    name=hist_names[i],
+                )
             )
 
             # set x-axis properties
             if xlim:
-                ax.set_xlim(xlim)
+                fig.update_xaxes(range=xlim)
```

Comment:
Tip to prevent code duplication. Rather than this pattern:
```
if a:
    function(a)
else:
   function(default_params)
```

You could also write

```
if not a:
   a = default_params

function(a)
```

As long as `a` is a local variable not being used later on, this will be more maintainable.